### ISSUE TITLE
Buggy output +warnings for "touching" polygons

### ISSUE THREAD

**Issue by AtiX** [w8r/martinez#10]
Hi,

I know touching polygons are probably considered an edge case, but I'd like to supply a test case where the clipping algorithm produces buggy results:

[Codepen](http://codepen.io/codepenAtiX/pen/mRPQXb)

**Situation:**
![grafik](https://cloud.githubusercontent.com/assets/236678/21899787/d56ddebc-d8f1-11e6-9f46-6059f1887535.png)
```
Big: [[0, 0], [3, 0], [3, 3], [0, 3], [0, 0]]
Small: [[3, 1], [4, 1], [4, 2], [3, 2], [3, 1]]
```

**First case**: Union produces a polygon which consist out of the merged surfaces (good), but also a degenerated polygon with only two points (less good):
![grafik](https://cloud.githubusercontent.com/assets/236678/21899832/069d9f4a-d8f2-11e6-8f9d-9cbd9381f0bb.png)
```
[
  [[0,0],[3,0],[3,1],[4,1],[4,2],[3,2],[3,3],[0,3],[0,0]],
  [[3,2],[3,3]]
]
```

**Second case**: Diff misses the upper left corner and produces a triangular polygon with a collinear point:
![grafik](https://cloud.githubusercontent.com/assets/236678/21900138/0c27cb92-d8f3-11e6-9c1f-b0018a18f701.png)
```
[
   [[0,0],[3,0],[3,1],[3,2],[3,3],[0,0]]
]
```

**Warning in all cases:** *"Edges of the same polygon overlap" [3, 1] [3, 3] [3, 2] [3, 3]* - this is neither the input data nor the output data - so probably some in between (internal) result?

From the polygon clipping libraries I tested so far, this one works best. How can I best help you resolve this issue?

**Comment by mandysimon88** [comment]
Arthur,

These are all part of the same problem, dealing with overlapping edges. There’s another open issue about that… same root problem though.

Amanda

**Comment by w8r** [comment]
This case is fixed in the upcoming `0.2.0` release
